daemon/algod/api/server/v2/utils.go

```diff
@@ -67,6 +67,14 @@ func notImplemented(ctx echo.Context, internal error, external string, log loggi
 	return returnError(ctx, http.StatusNotImplemented, internal, external, log)
 }
 
+func uint64MapKeys[M ~map[K]V, K ~uint64, V any](m M) []uint64 {
```

Comment:
Won't `maps.Keys()` work? Or maybe this coerces things to uint64, even if the keys are types like `basics.AppIndex` that have uint64 underlying types?

Reply:
If that's it, is it legal to make a generic "slice cast" kind of function, that takes a slice of `~uint64` and gives back `[]uint64`?  If so, then you could call that on `maps.Keys(m)` and the two things would be kept separate.

Reply:
Yes, `maps.Keys()` by itself won't work because it will return `[]basics.AppIndex` instead of `[]uint64`.

I took this suggestion in f343aa459